pkg/tekton/metapipeline/metapipeline.go

```diff
@@ -72,7 +73,18 @@ func CreateMetaPipelineCRDs(params CRDCreationParameters) (*tekton.CRDWrapper, e
 		return nil, err
 	}
 
-	resources := []*pipelineapi.PipelineResource{tekton.GenerateSourceRepoResource(params.PipelineName, params.GitInfo, "")}
+	revision := params.Branch
+	if params.PullRef != "" {
```

Comment:
I think this might be a bit too simple. Let me think about it for a bit. One thing we could do is to add a Revision option to `StepCreatePipelineOptions` and pass it through via `CRDCreationParameters`, but I'd like to spend a bit more time on this.

Reply:
Unrelated, but I think here is another issue - https://github.com/jenkins-x/jx/blob/fe66ed51204667e37ed067f65a7ebc67f7c75bbb/pkg/cmd/step/create/step_create_task.go#L248

we need to move the parsing of the pull refs higher up so that it is initialised in case cloning is needed.